testsuite/testplanet/project_test.c

```diff
@@ -9,11 +9,128 @@
 #include "uplink.h"
 
 void handle_project(UplinkProject *project);
+void test_revoke_access(UplinkProject *project);
+UplinkAccessResult derive_access(UplinkAccess *access);
+void test_access_availability(UplinkAccess *access, bool expect_available);
 
 int main()
 {
     with_test_project(&handle_project);
     return 0;
 }
 
-void handle_project(UplinkProject *project) { require(project->_handle != 0); }
+void handle_project(UplinkProject *project)
+{
+    require(project->_handle != 0);
+    test_revoke_access(project);
+}
+
+void test_revoke_access(UplinkProject *project)
+{
+    const char *access_string = getenv("UPLINK_0_ACCESS");
+
+    UplinkAccessResult access_result = uplink_parse_access(access_string);
+    require_noerror(access_result.error);
+    require(access_result.access != NULL);
+
+    UplinkBucketResult bucket_result = uplink_create_bucket(project, "alpha");
+    require_noerror(bucket_result.error);
+    uplink_free_bucket_result(bucket_result);
+
+    UplinkAccess *access = access_result.access;
+
+    {
+        UplinkAccessResult derived_access_result = derive_access(access);
+        test_access_availability(derived_access_result.access, true);
+
+        UplinkError *revoke_error = uplink_revoke_access(NULL, derived_access_result.access);
+        require_error(revoke_error, UPLINK_ERROR_INTERNAL);
+        uplink_free_error(revoke_error);
+
+        test_access_availability(derived_access_result.access, true);
+        uplink_free_access_result(derived_access_result);
+    }
+
+    {
+        UplinkAccessResult derived_access_result = derive_access(access);
+        test_access_availability(derived_access_result.access, true);
+
+        UplinkError *revoke_error = uplink_revoke_access(project, NULL);
+        require_error(revoke_error, UPLINK_ERROR_INTERNAL);
+        uplink_free_error(revoke_error);
+
+        test_access_availability(derived_access_result.access, true);
+        uplink_free_access_result(derived_access_result);
+    }
+
+    {
+        UplinkAccessResult derived_access_result = derive_access(access);
+
+        test_access_availability(derived_access_result.access, true);
+
+        UplinkError *revoke_error = uplink_revoke_access(project, derived_access_result.access);
+        require_noerror(revoke_error);
+        uplink_free_error(revoke_error);
+
+        test_access_availability(derived_access_result.access, false);
+        uplink_free_access_result(derived_access_result);
+    }
+
+    uplink_free_access_result(access_result);
+}
+
+UplinkAccessResult derive_access(UplinkAccess *access)
+{
+    UplinkPermission full_permissions = {
+        .allow_upload = true,
+        .allow_download = true,
+        .allow_delete = true,
+        .allow_list = true,
+    };
+
+    UplinkSharePrefix prefixes[] = {
+        {"alpha", ""},
```

Comment:
When I was writing tests, I referred to [this line](https://github.com/storj/uplink-c/blob/f103ceb38d70499974677398af78d492f00caa7a/testsuite/testplanet/access_test.c#L52) as an example of sharing access.
So I used the same bucket name and the prefix to be consistent with the rest of the codebase.

As far as I understood there are some issues [with prefix name](https://github.com/storj/uplink-c/blob/f103ceb38d70499974677398af78d492f00caa7a/testsuite/testplanet/access_test.c#L100) and because of it, the empty prefix was set.